ext/tls/tls_key.rs

```diff
@@ -0,0 +1,308 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+use crate::Certificate;
+use crate::PrivateKey;
+use deno_core::anyhow::anyhow;
+use deno_core::error::AnyError;
+use deno_core::futures::future::poll_fn;
+use deno_core::futures::future::Either;
+use deno_core::futures::FutureExt;
+use deno_core::unsync::spawn;
+use rustls::ServerConfig;
+use rustls_tokio_stream::ServerConfigProvider;
+use std::cell::RefCell;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::future::ready;
+use std::future::Future;
+use std::io::ErrorKind;
+use std::rc::Rc;
+use std::sync::Arc;
+use tokio::sync::broadcast;
+use tokio::sync::mpsc;
+use tokio::sync::oneshot;
+
+type ErrorType = Rc<AnyError>;
+
+/// A TLS certificate/private key pair.
```

Comment:
Can you add a module docstring explaining that this object is meant to be a GCable JS object?

Reply:
We don't actually rely on GC for these to get torn down in the normal case. It just happens to be easier to wrap the objects in cppgc rather than go through resources. Most of the time we'll create them and then immediately take them from the `TlsKeysHolder`.

I'll add some docs to describe how these get used.

Reply:
Thanks